src/locales/it/constants.ts

```diff
@@ -0,0 +1,230 @@
+    "novembre": 11,
+    "nov": 11,
+    "nov.": 11,
+    "dicembre": 12,
+    "dic": 12,
+    "dice": 12,
+    "dic.": 12,
+};
+
+export const INTEGER_WORD_DICTIONARY: { [word: string]: number } = {
+    'uno': 1,
+    'due': 2,
+    'tre': 3,
+    'quattro': 4,
+    'cinque': 5,
+    'sei': 6,
+    'sette': 7,
+    'otto': 8,
+    'nove': 9,
+    'dieci': 10,
+    'undici': 11,
+    'dodici': 12,
+};
+
+export const ORDINAL_WORD_DICTIONARY: { [word: string]: number } = {
+    "primo": 1,
+    "secondo": 2,
+    "terzo": 3,
+    'quarto': 4,
+    'quinto': 5,
+    'sesto': 6,
+    'settimo': 7,
+    'ottavo': 8,
+    'nono': 9,
+    'decimo': 10,
+    'undicesimo': 11,
+    'dodicesimo': 12,
+    'tredicesimo': 13,
+    'quattordicesimo': 14,
+    'quindicesimo': 15,
+    'sedicesimo': 16,
+    'diciassettesimo': 17,
+    'diciottesimo': 18,
+    'diciannovesimo': 19,
+    'ventesimo': 20,
+    "ventunesimo": 21,
+    "ventiduesimo": 22,
+    "ventitreesimo": 23,
+    "ventiquattresimo": 24,
+    "venticinquesimo": 25,
+    "ventiseiesimo": 26,
+    "ventisettesimo": 27,
+    "ventottesimo": 28,
+    "ventinovesimo": 29,
+    "trentesimo": 30,
+    "trentunesimo": 31,
+};
+
+export const TIME_UNIT_DICTIONARY: { [word: string]: OpUnitType | QUnitType } = {
+    "sec": "second",
+    "secondo": "second",
+    "secondi": "second",
+    "min": "minute",
+    "mins": "minute",
+    "minuti": "minute",
+    "h": "hour",
+    "hr": "hour",
+    "o": "hour",
+    "ora": "hour",
+    "ore": "hour",
+    "giorno": "d",
+    "giorni": "d",
+    "settimana": "week",
+    "settimane": "week",
+    "mese": "month",
+    "trimestre": "quarter",
+    "trimestri": "quarter",
+    "anni": "year",
+    "anno": "year",
+};
+
+//-----------------------------
+
+export const NUMBER_PATTERN = `(?:${matchAnyPattern(
+    INTEGER_WORD_DICTIONARY
+)}|[0-9]+|[0-9]+\\.[0-9]+|half(?:\\s{0,2}un?)?|un?\\b(?:\\s{0,2}qualcuno)?|qualcuno|molti|a?\\s{0,2}alcuni\\s{0,2}(?:of)?)`;
+
+export function parseNumberPattern(match: string): number {
+    const num = match.toLowerCase();
+    if (INTEGER_WORD_DICTIONARY[num] !== undefined) {
+        return INTEGER_WORD_DICTIONARY[num];
+    } else if (num === "un" || num === "una") {
+        return 1;
+    } else if (num.match(/alcuni/)) {
+        return 3;
+    } else if (num.match(/metá/)) {
+        return 0.5;
+    } else if (num.match(/paio/)) {
+        return 2;
+    } else if (num.match(/molti/)) {
+        return 7;
+    }
+
+    return parseFloat(num);
+}
+
+//-----------------------------
+
+export const ORDINAL_NUMBER_PATTERN = `(?:${matchAnyPattern(ORDINAL_WORD_DICTIONARY)}|[0-9]{1,2}(?:mo|ndo|rzo|simo|esimo)?)`;
+export function parseOrdinalNumberPattern(match: string): number {
+    let num = match.toLowerCase();
+    if (ORDINAL_WORD_DICTIONARY[num] !== undefined) {
+        return ORDINAL_WORD_DICTIONARY[num];
+    }
+
+    num = num.replace(/(?:imo|ndo|rzo|rto|nto|sto|tavo|nono|cimo|timo|esimo)$/i, "");
```

Comment:
It makes no sense since ordinal numbers in Italian are obtained by adding "°" sign (eg. 1°, 2°, 10°, 152°)